salt/proxy/restconf.py

```diff
@@ -0,0 +1,222 @@
+"""
+Proxy Minion to manage Restconf Devices
+
+:codeauthor: Jamie (Bear) Murphy <jamiemurphyit@gmail.com>
+:maturity:   new
+:platform:   any
+
+Usage
+=====
+
+.. note::
+
+    To be able to use this module you need to enable to RESTCONF on your device
+    and having https enabled.
+
+    Cisco Configuration example:
+
+    .. code-block:: bash
+
+        switch# conf t
+        switch(config)# restconf
+        switch(config)# ip http secure-server
+
+.. note::
+
+    RESTCONF requires modern OS distributions.
+    This plugin has been written specifically to use JSON Restconf endpoints
+
+Pillar
+------
+
+The ``restconf`` proxy configuration requires the following parameters in order
+to connect to the network switch:
+
+transport: ``https``
+    Specifies the type of connection transport to use. Valid values for the
+    connection are ``http``, and  ``https``.
+
+host: ``localhost``
+    The IP address or DNS host name of the connection device.
+
+username: ``admin``
+    The username to pass to the device to authenticate the RESTCONF requests.
+
+password:
+    The password to pass to the device to authenticate the RESTCONF requests.
+
+# TODO: timeout not yet implemented
+# timeout: ``60``
+#     Time in seconds to wait for the device to respond. Default: 60 seconds.
+
+verify: ``True``
+    Either a boolean, in which case it controls whether we verify the NX-API
+    TLS certificate, or a string, in which case it must be a path to a CA bundle
+    to use. Defaults to ``True``.
+
+    When there is no certificate configuration on the device and this option is
+    set as ``True`` (default), the commands will fail with the following error:
+    ``SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:581)``.
+    In this case, you either need to configure a proper certificate on the
+    device (*recommended*), or bypass the checks setting this argument as ``False``
+    with all the security risks considered as you may be MITM'd.
```

Comment:
**suggestion** use, I think it's `.. warning::` for the "False" option.

Reply:
thanks. added that and cleaned up